internal/v6provider/resource_load_balancer.go

```diff
@@ -0,0 +1,409 @@
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Required: true,
+						},
+					},
+				},
+			},
+			"tag": schema.ListNestedAttribute{
+				Optional: true,
+				Computed: true,
+				Validators: []validator.List{
+					listvalidator.ConflictsWith(
+						path.MatchRoot("virtual_machine"),
+						path.MatchRoot("virtual_machine_group"),
+					),
+				},
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Required: true,
+						},
+					},
+				},
+			},
+			"ip_address": schema.StringAttribute{
+				Computed: true,
+			},
+			"https_redirect": schema.BoolAttribute{
+				Computed: true,
+			},
+		},
+	}
+}
+
+func (r *LoadBalancerResource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+	var plan LoadBalancerResourceModel
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	name := r.M.UseOrGenerateName(plan.Name.ValueString())
+
+	t, ids := extractLoadBalancerResourceTypeAndIDs(&plan)
+	if t == "" {
+		t = core.VirtualMachinesResourceType
+	}
+
+	args := &core.LoadBalancerCreateArguments{
+		Name:         name,
+		ResourceType: t,
+		ResourceIDs:  &ids,
+		DataCenter:   r.M.DataCenterRef,
+	}
+
+	lb, _, err := r.M.Core.LoadBalancers.Create(
+		ctx, r.M.OrganizationRef, args,
+	)
+	if err != nil {
+		resp.Diagnostics.AddError("Load Balancer Create Error", err.Error())
+		return
+	}
+
+	if err := r.LoadBalancerRead(ctx, lb.ID, &plan, &resp.State); err != nil {
+		resp.Diagnostics.AddError("Load Balancer Read Error", err.Error())
+		return
+	}
+
+	diags = resp.State.Set(ctx, plan)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r *LoadBalancerResource) Read(
+	ctx context.Context,
+	req resource.ReadRequest,
+	resp *resource.ReadResponse,
+) {
+	state := &LoadBalancerResourceModel{}
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if err := r.LoadBalancerRead(
+		ctx,
+		state.ID.ValueString(),
+		state,
+		&resp.State,
+	); err != nil {
+		resp.Diagnostics.AddError("Load Balancer Read Error", err.Error())
+		return
+	}
+
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r *LoadBalancerResource) Update(
+	ctx context.Context,
+	req resource.UpdateRequest,
+	resp *resource.UpdateResponse,
+) {
+	var plan LoadBalancerResourceModel
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var state LoadBalancerResourceModel
+	diags = req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var id string
+	if !plan.ID.IsNull() && plan.ID.ValueString() != "" {
+		id = plan.ID.ValueString()
+	} else {
+		id = state.ID.ValueString()
+		plan.ID = state.ID
+	}
```

Comment:
I think this can now be replaced by:

```suggestion
	id := state.ID.ValueString()
```

Since nothing uses `plan.ID` anymore, and it is populated by the call to `LoadBalancerRead`.

After that though, I think this is good to go :)